core-backend/src/funcs.rs

```diff
@@ -453,50 +416,46 @@ where
     }
 
     pub fn send(
-        pid_value_ptr: u32,
-        payload_ptr: u32,
-        len: u32,
+        pid_value: ReadAs<HashWithValue>,
+        payload: ReadPayloadLimited,
```

Comment:
can we maybe do it somehow like ReadAs< PayloadLimited >?

Reply:
Certainly not as `ReadAs`, because reading a payload requires a length argument as well, `ReadAs` only works for structures with known size
The need for `ReadPayloadLimited` comes from the encapsulation of early length checking and the conversion of bytes into `Payload`